shapeshift/web
# feat: use quote endpoint for THORSwap data

## Description

Updates the THORSwapper trade quote logic to use data from the quote endpoint instead of hand-rolling calculations, including those relating to:

- slippage
- buy amount
- fees

## Pull Request Type


## Issue (if applicable)

Partially closes https://github.com/shapeshift/web/issues/4360 (we still need to replace our limit/memo logic, but that's for another PR)

## Risk

Reasonably high with respect to THORSwapper trades.

## Testing

- Check THORSwapper quote/trade values look as expected, ensuring RUNE and non-RUNE trades are checked
- Check that THORSwapper trades complete as expected

### Engineering

☝️

In the response to the `lcd/thorchain/quote/swap` request, check that `expected_amount_out`  is the roughly same as the "you get" amount. It won't quite match as we store fees in USD, and so we are converting to and from before the state layer, and so we lose some precision.

The "amount before fees" should be the "you get" amount (factoring slippage) + `outbound` and `affiliate` fee amounts in the `fees` object.

### Operations

☝️

## Screenshots (if applicable)

N/A
